client/extensions/woocommerce/app/products/product-form-variations-card.js

```diff
@@ -22,7 +22,21 @@ const ProductFormVariationsCard = ( {
 } ) => {
 	const handleToggle = () => {
 		if ( 'variable' !== product.type ) {
-			editProduct( product, { type: 'variable', dimensions: {} } );
+			// Reset any 'simple' product settings that have been filled out,
```

Comment:
Maybe save the values (if any) in component state and put them back if they toggle it off again?